In [1]:
import numpy as np
import pandas as pd

In [2]:
df = pd.read_csv('geographic-sum-per-day-ver_00162.csv')

In [4]:
df.describe()

town_code      agas_code
count  658623.000000  554609.000000
mean     4830.878240     297.047244
std      2939.418622     417.479858
min        26.000000       1.000000
25%      2610.000000      18.000000
50%      5000.000000     127.000000
75%      7400.000000     423.000000
max      9800.000000    2911.000000

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 658623 entries, 0 to 658622
Data columns (total 18 columns):
 #   Column                               Non-Null Count   Dtype  
---  ------                               --------------   -----  
 0   town_code                            658623 non-null  int64  
 1   agas_code                            554609 non-null  float64
 2   town                                 658623 non-null  object 
 3   date                                 658623 non-null  object 
 4   accumulated_cases                    658623 non-null  object 
 5   new_cases_on_date                    658623 non-null  bool   
 6   accumulated_recoveries               658623 non-null  object 
 7   new_recoveries_on_date               658623 non-null  bool   
 8   accumulated_hospitalized             658623 non-null  object 
 9   new_hospitalized_on_date             658623 non-null  bool   
 10  accumulated_deaths                   658623 non-null  object 
 11  new_deaths_on

In [6]:
df.isna().sum()

town_code                                   0
agas_code                              104014
town                                        0
date                                        0
accumulated_cases                           0
new_cases_on_date                           0
accumulated_recoveries                      0
new_recoveries_on_date                      0
accumulated_hospitalized                    0
new_hospitalized_on_date                    0
accumulated_deaths                          0
new_deaths_on_date                          0
accumulated_diagnostic_tests                0
new_diagnostic_tests_on_date                0
accumulated_vaccination_first_dose          0
new_vacc_first_dose_on_date                 0
accumulated_vaccination_second_dose         0
new_vacc_second_dose_on_date                0
dtype: int64

In [25]:
for h in df.columns:
    print(h, "-----:\n", df[h].unique(), '\n')

town_code -----:
 [  26   28   31   41   46   47   53   70   99  122  147  154  166  168
  171  182  190  195  229  240  246  276  326  351  466  469  472  473
  478  480  481  482  483  485  487  489  490  492  494  496  498  499
  502  504  505  507  508  509  510  511  512  516  517  518  520  521
  522  524  525  526  528  529  530  531  532  534  535  536  537  538
  541  542  543  565  577  587  627  628  633  634  635  637  638  652
  654  658  666  681  696  709  763  812  831  874  913  922  944  962
  975  978  998 1015 1020 1031 1034 1050 1054 1059 1060 1061 1063 1066
 1082 1098 1113 1137 1139 1161 1167 1186 1192 1200 1224 1243 1245 1247
 1263 1268 1271 1274 1284 1286 1290 1292 1295 1296 1303 1304 1309 1310
 1315 1319 1326 1327 1330 1333 1337 1345 1347 1348 1358 1359 2034 2038
 2100 2200 2300 2400 2500 2530 2550 2560 2600 2610 2620 2630 2640 2650
 2660 2710 2720 2730 2800 3000 3557 3560 3563 3564 3570 3574 3576 3604
 3611 3616 3617 3618 3638 3640 3641 3647 3650 3652 3660 370

In [26]:
df.town_code.value_counts()

3000    58756
5000    44464
4000    28187
8300    22629
7900    19850
        ...  
3726      397
658       397
147       397
2710      397
511       397
Name: town_code, Length: 262, dtype: int64

In [27]:
df.town.value_counts()

ירושלים        58756
תל אביב-יפו    44464
חיפה           28187
ראשון לציון    22629
חולון          19850
               ...  
אפרת             397
תל מונד          397
רמת ישי          397
ג'לג'וליה        397
כפר כמא          397
Name: town, Length: 262, dtype: int64

In [28]:
df.agas_code.value_counts()

3.0       16277
2.0       15880
4.0       14292
1.0       12307
5.0       11910
          ...  
1043.0      397
616.0       397
1216.0      397
335.0       397
347.0       397
Name: agas_code, Length: 337, dtype: int64

In [39]:
jlm_df = df.loc[df['town_code'] == 3000]

In [36]:
jlm_df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 58756 entries, 195324 to 254079
Data columns (total 18 columns):
 #   Column                               Non-Null Count  Dtype  
---  ------                               --------------  -----  
 0   town_code                            58756 non-null  int64  
 1   agas_code                            58359 non-null  float64
 2   town                                 58756 non-null  object 
 3   date                                 58756 non-null  object 
 4   accumulated_cases                    58756 non-null  object 
 5   new_cases_on_date                    58756 non-null  bool   
 6   accumulated_recoveries               58756 non-null  object 
 7   new_recoveries_on_date               58756 non-null  bool   
 8   accumulated_hospitalized             58756 non-null  object 
 9   new_hospitalized_on_date             58756 non-null  bool   
 10  accumulated_deaths                   58756 non-null  object 
 11  new_deaths_on_date    

### Note we have ~ 59000 samples from Jerusalem

In [37]:
jlm_df.isna().sum()

town_code                                0
agas_code                              397
town                                     0
date                                     0
accumulated_cases                        0
new_cases_on_date                        0
accumulated_recoveries                   0
new_recoveries_on_date                   0
accumulated_hospitalized                 0
new_hospitalized_on_date                 0
accumulated_deaths                       0
new_deaths_on_date                       0
accumulated_diagnostic_tests             0
new_diagnostic_tests_on_date             0
accumulated_vaccination_first_dose       0
new_vacc_first_dose_on_date              0
accumulated_vaccination_second_dose      0
new_vacc_second_dose_on_date             0
dtype: int64

### Note that we have 397 samples with statistical zone set to N/A in Jerusalem

In [41]:
jlm_df.agas_code.value_counts()

1131.0    397
1355.0    397
1212.0    397
825.0     397
1014.0    397
         ... 
134.0     397
111.0     397
132.0     397
915.0     397
1015.0    397
Name: agas_code, Length: 147, dtype: int64

### We have 146 different statistical zones in JLM